tests/tt_metal/tt_metal/unit_tests/common/device_fixture.hpp

```diff
@@ -33,9 +33,11 @@ class DeviceFixture : public ::testing::Test {
             auto* device = tt::tt_metal::CreateDevice(id);
             devices_.push_back(device);
         }
```

Comment:
@abhullar-tt calling these cluster enable/disable of the internal routing

Reply:
can this api be hidden from user and be called when the umd for a mmio device + devices its connected to via ethernet is created or would setting this to `true` break SD flow?

Reply:
ahh yeah sorry rmr this discussion, do you think it makes sense for remote device initialization to call this?

Reply:
yea probably. Wherever we will use this CreateDevice pattern that is used in gtests.

Reply:
nvm my original comment, if each remote device calls this api it may mess up the state for other already initialized devices

Reply:
@TT-billteng we'll need (probably) a metal host api that does this:
```
void initialize_my_cluster(vec<Devices> devs) {
    find all required mmio chips and remote chips to support all devs
    for ( all required chips) {
        CreateDevice(required chip);
    }
    tt::Cluster::instance().set_internal_routing_info_for_ethernet_cores(true);
}
```

The main reason is that users may only want to use one remote chip, but at the metal level we need to create and initialize some other devices to support a FD tunneling path to that remote chip.

Reply:
yeah I think this is a good idea and it can accept target device ids

Reply:
are you saying this is a top-level API in `host_api.hpp`?

sounds like if had a device pool that's always initialized under the hood, this wouldn't need to be another top-level API

Reply:
I think that is the same discussion as "why do we have users call CreateDevice for single chip today". We are proposing some api that is hot-swappable with the CreateDevice call for single chip.

Reply:
if the user needs to only use one chip, I feel like it would be confusing for them to have to figure out which other chips they need to start up to get the remote chip working. This should be infra hidden under the top-level API

Reply:
I can imagine we'd want an API that creates a handle to more than 1 or all the devices like `CreateDevices( vector<Device*> dev)` but if user wants access to 1 remote chip only, we should figure out the associated MMIO chip to start up as well

Reply:
Yea, Almeet and I agree with that. We have to find the user all the chips to initialize, because the FD routing is something tt_metal manages under the hood.

Reply:
tracked here: https://github.com/tenstorrent-metal/tt-metal/issues/5048